src/server/Server.ts

```diff
@@ -16,8 +16,23 @@ import Logger from '@/utils/Logger';
 import CommandManager from '@/command/CommandManager';
 import RakNet from '@/network/rakNet/RakNet';
 import PacketManager from '@/network/bedrock/PacketManager';
+import { ServerOptions } from './ServerOptions'
 
-class Server extends EventEmitter {
+class Server extends EventEmitter implements ServerOptions {
+    ServerName = "NetrexTestServer"
```

Comment:
No use of public or private deceleration, needs changing.
Issues:
- Declared as class constants with this case
- Should be initialized with passed options
- PVP will be handled by plugins
- RCON will be its separate plugin
- Hardcore, not sure what you mean by this

Reply:
k bro